<?xml version="1.0" encoding="UTF-8"?>
<testng-results skipped="0" failed="3" total="10" passed="7">
  <reporter-output>
  </reporter-output>
  <suite name="Suite" duration-ms="1935" started-at="2021-07-09T04:32:36Z" finished-at="2021-07-09T04:32:38Z">
    <groups>
    </groups>
    <test name="CycleCheck" duration-ms="94" started-at="2021-07-09T04:32:36Z" finished-at="2021-07-09T04:32:36Z">
      <class name="BasicTests.CycleCheck">
        <test-method status="FAIL" signature="cycle_check()[pri:0, instance:BasicTests.CycleCheck@2ab4bc72]" name="cycle_check" duration-ms="79" started-at="2021-07-09T10:02:36Z" finished-at="2021-07-09T10:02:36Z">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Architecture Violation [Priority: MEDIUM] - Rule 'slices matching '..demo..(*)..' should be free of cycles' was violated (1 times):
Cycle detected: Slice firstservice -> 
                Slice secondservice -> 
                Slice firstservice
  1. Dependencies of Slice firstservice
    - Field <demo.service.firstservice.FirstService.obj> has type <demo.service.secondservice.SecondService> in (FirstService.java:0)
  2. Dependencies of Slice secondservice
    - Field <demo.service.secondservice.SecondService.obj> has type <demo.service.firstservice.FirstService> in (SecondService.java:0)]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Architecture Violation [Priority: MEDIUM] - Rule 'slices matching '..demo..(*)..' should be free of cycles' was violated (1 times):
Cycle detected: Slice firstservice -> 
                Slice secondservice -> 
                Slice firstservice
  1. Dependencies of Slice firstservice
    - Field <demo.service.firstservice.FirstService.obj> has type <demo.service.secondservice.SecondService> in (FirstService.java:0)
  2. Dependencies of Slice secondservice
    - Field <demo.service.secondservice.SecondService.obj> has type <demo.service.firstservice.FirstService> in (SecondService.java:0)
	at com.tngtech.archunit.lang.ArchRule$Assertions.assertNoViolation(ArchRule.java:94)
	at com.tngtech.archunit.lang.ArchRule$Assertions.check(ArchRule.java:82)
	at com.tngtech.archunit.lang.ArchRule$Factory$SimpleArchRule.check(ArchRule.java:198)
	at com.tngtech.archunit.lang.syntax.ObjectsShouldInternal.check(ObjectsShouldInternal.java:81)
	at com.tngtech.archunit.library.dependencies.SliceRule.check(SliceRule.java:64)
	at BasicTests.CycleCheck.cycle_check(CycleCheck.java:26)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:659)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:845)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1153)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:771)
	at org.testng.TestRunner.run(TestRunner.java:621)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:357)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:352)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:310)
	at org.testng.SuiteRunner.run(SuiteRunner.java:259)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1199)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1124)
	at org.testng.TestNG.run(TestNG.java:1032)
	at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- cycle_check -->
      </class> <!-- BasicTests.CycleCheck -->
    </test> <!-- CycleCheck -->
    <test name="DependencyCheck" duration-ms="31" started-at="2021-07-09T04:32:36Z" finished-at="2021-07-09T04:32:36Z">
      <class name="BasicTests.DependencyCheck">
        <test-method status="PASS" signature="entity_class_should_not_depend_on_anyone()[pri:0, instance:BasicTests.DependencyCheck@147a5d08]" name="entity_class_should_not_depend_on_anyone" duration-ms="29" started-at="2021-07-09T10:02:36Z" finished-at="2021-07-09T10:02:36Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- entity_class_should_not_depend_on_anyone -->
      </class> <!-- BasicTests.DependencyCheck -->
    </test> <!-- DependencyCheck -->
    <test name="LayerAcess" duration-ms="10" started-at="2021-07-09T04:32:38Z" finished-at="2021-07-09T04:32:38Z">
      <class name="BasicTests.LayerArchitecture">
        <test-method status="PASS" signature="layer_architecture()[pri:0, instance:BasicTests.LayerArchitecture@22295ec4]" name="layer_architecture" duration-ms="9" started-at="2021-07-09T10:02:38Z" finished-at="2021-07-09T10:02:38Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- layer_architecture -->
      </class> <!-- BasicTests.LayerArchitecture -->
    </test> <!-- LayerAcess -->
    <test name="NamingConvention" duration-ms="7" started-at="2021-07-09T04:32:38Z" finished-at="2021-07-09T04:32:38Z">
      <class name="BasicTests.NamingConvention">
        <test-method status="FAIL" signature="controllers_should_be_suffixed()[pri:0, instance:BasicTests.NamingConvention@bff34c6]" name="controllers_should_be_suffixed" duration-ms="2" started-at="2021-07-09T10:02:38Z" finished-at="2021-07-09T10:02:38Z">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Architecture Violation [Priority: MEDIUM] - Rule 'classes that reside in a package '..controller..' should have simple name ending with 'Controller'' was violated (1 times):
simple name of demo.controller.demo does not end with 'Controller' in (demo.java:0)]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Architecture Violation [Priority: MEDIUM] - Rule 'classes that reside in a package '..controller..' should have simple name ending with 'Controller'' was violated (1 times):
simple name of demo.controller.demo does not end with 'Controller' in (demo.java:0)
	at com.tngtech.archunit.lang.ArchRule$Assertions.assertNoViolation(ArchRule.java:94)
	at com.tngtech.archunit.lang.ArchRule$Assertions.check(ArchRule.java:82)
	at com.tngtech.archunit.lang.ArchRule$Factory$SimpleArchRule.check(ArchRule.java:198)
	at com.tngtech.archunit.lang.syntax.ObjectsShouldInternal.check(ObjectsShouldInternal.java:81)
	at BasicTests.NamingConvention.controllers_should_be_suffixed(NamingConvention.java:23)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:659)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:845)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1153)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:771)
	at org.testng.TestRunner.run(TestRunner.java:621)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:357)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:352)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:310)
	at org.testng.SuiteRunner.run(SuiteRunner.java:259)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1199)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1124)
	at org.testng.TestNG.run(TestNG.java:1032)
	at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- controllers_should_be_suffixed -->
        <test-method status="PASS" signature="interfaces_with_repository_should_be_in_a_repository_package()[pri:0, instance:BasicTests.NamingConvention@bff34c6]" name="interfaces_with_repository_should_be_in_a_repository_package" duration-ms="2" started-at="2021-07-09T10:02:38Z" finished-at="2021-07-09T10:02:38Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- interfaces_with_repository_should_be_in_a_repository_package -->
        <test-method status="PASS" signature="service_should_be_infix_with_service_word()[pri:0, instance:BasicTests.NamingConvention@bff34c6]" name="service_should_be_infix_with_service_word" duration-ms="1" started-at="2021-07-09T10:02:38Z" finished-at="2021-07-09T10:02:38Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- service_should_be_infix_with_service_word -->
      </class> <!-- BasicTests.NamingConvention -->
    </test> <!-- NamingConvention -->
    <test name="LocationChecks" duration-ms="25" started-at="2021-07-09T04:32:38Z" finished-at="2021-07-09T04:32:38Z">
      <class name="BasicTests.LocationCheck">
        <test-method status="PASS" signature="classes_annoated_with_controller_annotation_should_resides_in_controller_package()[pri:0, instance:BasicTests.LocationCheck@741b3bc3]" name="classes_annoated_with_controller_annotation_should_resides_in_controller_package" duration-ms="14" started-at="2021-07-09T10:02:38Z" finished-at="2021-07-09T10:02:38Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- classes_annoated_with_controller_annotation_should_resides_in_controller_package -->
        <test-method status="FAIL" signature="classes_annoated_with_entity_annotation_should_resides_in_entity_package()[pri:0, instance:BasicTests.LocationCheck@741b3bc3]" name="classes_annoated_with_entity_annotation_should_resides_in_entity_package" duration-ms="3" started-at="2021-07-09T10:02:38Z" finished-at="2021-07-09T10:02:38Z">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Architecture Violation [Priority: MEDIUM] - Rule 'classes that are annotated with @Entity should reside in a package '..entity..'' was violated (1 times):
Class <demo.controller.demo> does not reside in a package '..entity..' in (demo.java:0)]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Architecture Violation [Priority: MEDIUM] - Rule 'classes that are annotated with @Entity should reside in a package '..entity..'' was violated (1 times):
Class <demo.controller.demo> does not reside in a package '..entity..' in (demo.java:0)
	at com.tngtech.archunit.lang.ArchRule$Assertions.assertNoViolation(ArchRule.java:94)
	at com.tngtech.archunit.lang.ArchRule$Assertions.check(ArchRule.java:82)
	at com.tngtech.archunit.lang.ArchRule$Factory$SimpleArchRule.check(ArchRule.java:198)
	at com.tngtech.archunit.lang.syntax.ObjectsShouldInternal.check(ObjectsShouldInternal.java:81)
	at BasicTests.LocationCheck.classes_annoated_with_entity_annotation_should_resides_in_entity_package(LocationCheck.java:30)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:659)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:845)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1153)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:771)
	at org.testng.TestRunner.run(TestRunner.java:621)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:357)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:352)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:310)
	at org.testng.SuiteRunner.run(SuiteRunner.java:259)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1199)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1124)
	at org.testng.TestNG.run(TestNG.java:1032)
	at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- classes_annoated_with_entity_annotation_should_resides_in_entity_package -->
        <test-method status="PASS" signature="classes_annoated_with_service_annotation_should_resides_in_service_package()[pri:0, instance:BasicTests.LocationCheck@741b3bc3]" name="classes_annoated_with_service_annotation_should_resides_in_service_package" duration-ms="2" started-at="2021-07-09T10:02:38Z" finished-at="2021-07-09T10:02:38Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- classes_annoated_with_service_annotation_should_resides_in_service_package -->
        <test-method status="PASS" signature="interfaces_should_not_placed_in_entity_package()[pri:0, instance:BasicTests.LocationCheck@741b3bc3]" name="interfaces_should_not_placed_in_entity_package" duration-ms="1" started-at="2021-07-09T10:02:38Z" finished-at="2021-07-09T10:02:38Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- interfaces_should_not_placed_in_entity_package -->
      </class> <!-- BasicTests.LocationCheck -->
    </test> <!-- LocationChecks -->
  </suite> <!-- Suite -->
</testng-results>
